### PR TITLE
Map visualization

### LINKED ISSUE #736: Visualization: Geo Maps
### Summary
Integration of the [deck.gl](https://deck.gl) which allows interactive maps visualizations with a lot of possible, predefined layers, like heat maps, transportation paths, and more. [Examples](https://deck.gl/examples):
image

### Specification
- The visualization should use the [Using @deck.gl/core](https://deck.gl/docs/get-started/using-standalone), which in contrast to the `scripting api` provides a declarative API, which is exactly what we need. [Be sure to check the declarative playground](https://deck.gl/playground)! For more information check out the implementation, especially the [setProps method](https://github.com/visgl/deck.gl/blob/8.2-release/modules/core/src/lib/deck.js#L243).
- In order to use Mapbox (used internally by deck.gl), we need an account and token key. Create an Enso account using `contact@enso.org` email. @iamrecursion will help with it.
- The visualization input data should be JSON containing a description of the location, all effects, all layers, their parameters, etc. An example config is provided below (please note that location parameters are using different layout than the one provided by the `Deck` class):
   ```js
   {
     latitude: 37.8,
     longitude: -122.45,
     zoom: 15,
     controller: true,
     layers: [
       new ScatterplotLayer({
         data: [
           {position: [-122.45, 37.8], color: [255, 0, 0], radius: 100}
         ],
         getColor: d => d.color,
         getRadius: d => d.radius
       })
      ]
   }
   ```
   
   Of course, a shorter version should work as well:
   ```js
   {
     latitude: 37.8,
     longitude: -122.45
   }
   ```

- The visualization should be implemented as a JavaScript visualization, according to [the JavaScript visualization type documentation](https://github.com/enso-org/ide/blob/main/src/rust/ide/lib/view/src/graph_editor/component/visualization/foreign/java_script/definition.rs) and [here](https://dev.enso.org/docs/ide/product/visualizations.html#custom-visualization-example).
- The visualization should be defined as a `JavaScript` file and should be placed in the right place, according to the engine specification - it should be an example visualization implementation, not an embedded visualization placed in Rust codebase!.

### Acceptance Criteria & Test Cases
- Working visualization in IDE - should display a map for the given JSON input data.
- The following configurations should be tested and saved in some directory (not in IDE repo) as example JSON configs. Please note that this does NOT include any visible sliders in GUI changing the parameters. All parameters should be controlled only by the JSON data:
    - Map zoomed to San Francisco
    - [This example](https://deck.gl/examples/arc-layer)
    - [This example](https://deck.gl/examples/geojson-layer-polygons)
    - [This example](https://deck.gl/examples/geojson-layer-paths)
    - [This example](https://deck.gl/examples/heatmap-layer)
    - [This example](https://deck.gl/examples/hexagon-layer)
    - [This example](https://deck.gl/examples/trips-layer)
- It should be possible to interact (zoom, pan, rotate) with the visualization using the touchpad and mouse wheel, after making the visualization active (after clicking the visualization). Making visualizations active is a part of another issue, but testing it in IDE is part of this issue and this issue should not be closed until it works 100% correctly.
- The visualization controls should be the same as our graphs (Mapbox exposes API to change that):
   - Panning should be done by two finger swipe gesture
   - Zooming should be done by pinching 
   - Clicking and dragging should rotate the view.
- @sylwiabr  should confirm that the visualization behaves correctly before merging this issue to the Main branch.

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description

Zrzut ekranu 2020-09-7 o 15 59 24

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md).
- [x] All code has been tested where possible.
- [ ] All code has been profiled where possible.

### CODE REVIEW

File: src/js/lib/content/src/index.html
Comment (farmaazon) [inline src/js/lib/content/src/index.html:45]: Could it be somehow added by loaded visualization itself, instead of embedding it into index.html?

**Review by farmaazon — COMMENTED** [review]
I would wait for @wdanilo view review as well.

**Review by wdanilo — CHANGES_REQUESTED** [review]
image

**Review by wdanilo — APPROVED** [review]

**Comment by wdanilo** [comment]
Please lets talk about 
image
before merging

**Review by farmaazon — COMMENTED** [review]
Please add the examples for using maps as an issue comment (according to its description).

File: src/js/lib/content/src/index.html
Comment (BinarySoftware) [reply src/js/lib/content/src/index.html]: We've talked about that on the phone and agreed what to do, the correct config is already on main.

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js
Comment (BinarySoftware) [reply src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js:14]: As far as i can tell it downloads and attaches the scripts already while launching the app, and there is only one script when i show/hide vis multiple times, so this is not the case.

**Comment by BinarySoftware** [comment]
Here are the screens:
Zrzut ekranu 2020-09-22 o 16 14 10

Zrzut ekranu 2020-09-22 o 16 11 18
Zrzut ekranu 2020-09-22 o 16 09 12

**Pull request merged**
